bibo/internals.py

```diff
@@ -192,3 +195,50 @@ def complete_path(ctx, args, incomplete):
         if os.access(path, os.R_OK):
             options.append((path, os.path.basename(path)))
     return options
+
+
+def match_case(substring, target):
+    """
+    Return the string `substring` with the same case as in the `target`
+    string.
+    """
+    index = target.lower().find(substring.lower())
+    if index < 0:
+        msg = "Failed to match case: {} not in {}".format(substring, target)
+        raise ValueError(msg)
+    return target[index : (index + len(substring))]
+
+
+_ansi_bold = "\033[1m"
+_ansi_unbold = "\033[22m"
+
+
+def highlight_text(text, highlight):
+    """
+    Return `text` with sub-string `highlight` in bold.
+    """
+    text = re.sub(
+        highlight, "{}{}{}".format(_ansi_bold, highlight, _ansi_unbold), text,
+    )
+    # Drop ANSI unbold followed by ANSI bold
+    return text.replace(_ansi_unbold + _ansi_bold, "")
```

Comment:
Why is this necessary? highlight == '', or quirk of re-sub? If that's the case, I suppose it should return out earlier.

Reply:
The main reasons I think was to write more straightforward tests. You want to compare the string after `highlight_text` with something you make yourself in the test case that looks meaningful. Was also thinking about suggesting some patches to `click` itself with smarter styling. Would you like to test how bibo behaves without this line?